### PR TITLE
Add part 3 of FFI tutorial

### PR DESCRIPTION
**Pull request by takikawa**
Part 3! Any feedback appreciated.

### CODE REVIEW

File: blog/_src/posts/2016-07-11-tutorial-racket-ffi-part-3.scrbl [hunk 1]
```diff
@@ -0,0 +1,504 @@
...
+  (cast (list 1.3 5.8) (_list-struct _double _double) _cairo_path_data_t))
+a-union-val
+]
+
+These two lines first construct a union object (via the @racket[_cairo_path_data_t]
+type) using a @racket[cast]. A cast is just an operation that lets you coerce
+from one C type to another. We use it in this example since it's just an easy way
+to generate a union object.
+
+The second line just shows that a union prints as an opaque object. You can't do
```
Comment (bennn) [inline blog/_src/posts/2016-07-11-tutorial-racket-ffi-part-3.scrbl:196]: ~~just~~ (here and L193 maybe -- too many "just"!)

File: blog/_src/posts/2016-07-11-tutorial-racket-ffi-part-3.scrbl [hunk 2]
```diff
@@ -0,0 +1,504 @@
...
+The first thing the function does is extract the pointer out of the
+representation, and then immediately calls @racket[ptr-ref] on it. This
+lets us manipulate the C path struct using the simple representation we
+defined in the first part of the blog post.
+
+Note: in case you're not very familiar with Racket pattern matching, the
+@racket[match-define] form is a very convenient form that lets you define
+potentially multiple variables using a pattern. The first argument clause
+is a pattern and the second is an expression to match on. Check it out
+in the docs.
```
Comment (bennn) [inline blog/_src/posts/2016-07-11-tutorial-racket-ffi-part-3.scrbl:415]: http://docs.racket-lang.org/reference/match.html#%28form._%28%28lib._racket%2Fmatch..rkt%29._match-define%29%29

File: blog/_src/posts/2016-07-11-tutorial-racket-ffi-part-3.scrbl [hunk 3]
```diff
@@ -0,0 +1,504 @@
...
+]
+
+The first thing the function does is extract the pointer out of the
+representation, and then immediately calls @racket[ptr-ref] on it. This
+lets us manipulate the C path struct using the simple representation we
+defined in the first part of the blog post.
+
+Note: in case you're not very familiar with Racket pattern matching, the
+@racket[match-define] form is a very convenient form that lets you define
+potentially multiple variables using a pattern. The first argument clause
```
Comment (bennn) [inline blog/_src/posts/2016-07-11-tutorial-racket-ffi-part-3.scrbl:413]: Suggestion: "... the `@racket[match-define]` form lets you define potentially multiple variables using a pattern, similar to Haskell or OCaml's `@racket[let]` statement."

File: blog/_src/posts/2016-07-11-tutorial-racket-ffi-part-3.scrbl [hunk 4]
```diff
@@ -0,0 +1,504 @@
...
+              (displayln elem))
+            (cairo-stroke ctx)))
+]
+
+Notice how the sequence prints out as an intuitive list of commands
+instead of a bunch of opaque union values as we saw before when using
+the @racket[_array/list] type.
+
+That concludes part 3 of the FFI tutorial. Hopefully you're now equipped
+to deal with union types and custom C types. If not, see the FFI reference
```
Comment (bennn) [inline blog/_src/posts/2016-07-11-tutorial-racket-ffi-part-3.scrbl:503]: union types: http://docs.racket-lang.org/foreign/C_Union_Types.html

File: blog/_src/posts/2016-07-11-tutorial-racket-ffi-part-3.scrbl [hunk 5]
```diff
@@ -0,0 +1,504 @@
...
+for more details and examples.
```
Comment (bennn) [inline blog/_src/posts/2016-07-11-tutorial-racket-ffi-part-3.scrbl:504]: FFI reference: http://docs.racket-lang.org/foreign/index.html

File: blog/_src/posts/2016-07-11-tutorial-racket-ffi-part-3.scrbl [hunk 6]
```diff
@@ -0,0 +1,504 @@
...
+  (_fun _cairo_t -> _cairo_path_t))
+
+(do-cairo (λ (ctx)
+            (cairo-move-to ctx 0.0 0.0)
+            (cairo-line-to ctx 100.0 100.0)
+            (define path (cairo-copy-path ctx))
+            (code:comment "Using path as a sequence")
+            (for ([elem path])
+              (displayln elem))
+            (cairo-stroke ctx)))
```
Comment (bennn) [inline blog/_src/posts/2016-07-11-tutorial-racket-ffi-part-3.scrbl:495]: Do more than just a line here? Maybe:

```
(do-cairo (λ (ctx)
              (cairo-move-to ctx 50.0 50.0)
              (cairo-line-to ctx 206.0 206.0)
              (cairo-move-to ctx 50.0 206.0)
              (cairo-line-to ctx 115.0 115.0)
              (define path (cairo-copy-path ctx))
              ; Using path as a sequence
              (for ([elem path])
                (displayln elem))
              (cairo-stroke ctx)))
```

**Comment by bennn** [comment]
great!

File: blog/_src/posts/2016-07-11-tutorial-racket-ffi-part-3.scrbl (hunk 1 above)
Comment (takikawa) [inline blog/_src/posts/2016-07-11-tutorial-racket-ffi-part-3.scrbl:196]: Good point (also ended up deleting a bunch of other "just"s).

**Comment by takikawa** [comment]
Thanks! Now merged.

**Pull request closed without merging**